### PR TITLE
fix(cloudflare_worker): add missing js module option to fix upload

### LINKED ISSUE #1417: Support for `cloudflare_worker_script` resources being module syntax
### Current Terraform and Cloudflare provider version

```
$ terraform -v
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v3.8.0
```

### Description

Similar to the proposed upstream change (cloudflare/cloudflare-go#794), `cloudflare_worker_script` resource needs to be aware of Worker module syntax. The quick tl;dr: Workers only accepts module-based workers when uploaded with `Content-Type: application/javascript+module` header.

It seems most worth it to add a `module` boolean parameter, and pass through through to cloudflare-go.

Attempting to upload a valid worker without this sort of change results in a response: 
```json
{
  "code": 10021,
  "message": "Uncaught SyntaxError: Unexpected token 'export'\n  at line 1\n"
}
```

### Use cases

Uploading workers scripts based on ES Modules, documented here: https://developers.cloudflare.com/workers/learning/migrating-to-module-workers

### Potential Terraform configuration

```hcl
resource "cloudflare_worker_script" "hello" {
  name    = "hello"
  content = "export default { async fetch() { return new Response('hello-world') } }"
  module  = true
}
```

### References

- cloudflare/cloudflare-go#794

### PR DESCRIPTION
**Pull request by js0cha**
Closes #1417

### CODE REVIEW

**Comment by jacobbednarz** [comment]
can we please add some acceptance tests to cover this updated behaviour? i'd like to make sure we have the coverage in place for this given it required a few updates to the core request/response lifecycle.

File: internal/provider/resource_cloudflare_worker_script.go [hunk 1]
```diff
@@ -129,6 +129,11 @@ func resourceCloudflareWorkerScriptCreate(ctx context.Context, d *schema.Resourc
 		return diag.FromErr(fmt.Errorf("script content cannot be empty"))
 	}
 
+	moduleEnabled := d.Get("module_enabled").(bool)
```
Comment (Electroid) [inline internal/provider/resource_cloudflare_worker_script.go:132]: I'd recommend `type`: either `"service-worker"` (default) or `"modules"`.

Comment (js0cha) [reply internal/provider/resource_cloudflare_worker_script.go:132]: I was think something about like `esm-enabled: true` or  just `esm: true`

Comment (jacobbednarz) [reply internal/provider/resource_cloudflare_worker_script.go:132]: let's match the go library and use `module: true`

**Comment by jacobbednarz** [comment]
i've fixed the docs + added a test for this which is now all passing

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareWorkerScript_" -count 1 -parallel 1 -timeout 120m -parallel 1
?       github.com/cloudflare/terraform-provider-cloudflare     [no test files]
=== RUN   TestAccCloudflareWorkerScript_Import
--- PASS: TestAccCloudflareWorkerScript_Import (12.91s)
=== RUN   TestAccCloudflareWorkerScript_MultiScriptEnt
=== PAUSE TestAccCloudflareWorkerScript_MultiScriptEnt
=== RUN   TestAccCloudflareWorkerScript_ModuleUpload
=== PAUSE TestAccCloudflareWorkerScript_ModuleUpload
=== CONT  TestAccCloudflareWorkerScript_MultiScriptEnt
--- PASS: TestAccCloudflareWorkerScript_MultiScriptEnt (35.64s)
=== CONT  TestAccCloudflareWorkerScript_ModuleUpload
--- PASS: TestAccCloudflareWorkerScript_ModuleUpload (9.73s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/provider   58.849s
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check   [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check        [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check      [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/version     [no test files]
```

**Comment by jacobbednarz** [comment]
thanks for this one @js0cha 👏

**Pull request merged**